src/main/scala/com/kinja/amqp/AmqpConfiguration.scala

```diff
@@ -26,6 +26,8 @@ final case class ResendLoopConfig(
 	memoryFlushTimeOut: FiniteDuration
 )
 
+final case class ProducerConfig(guaranteedDelivery: Boolean, exchangeParams: ExchangeParameters)
```

Comment:
I think it would provide a more comprehensible API if you defined delivery guarantee as an enum instead of a boolean, e.g. `deliveryGuarantee = AtLeastOnce | AtMostOnce`